### PR TITLE
i#2868 test failures: checklevel only in debug

### LINKED ISSUE #2868: Test failures with gcc 6 and recent Linux distro
On a recently upgraded Linux machine these tests are now failing:

```
	 21 - code_api|common.nativeexec_exe (Failed)
	 22 - code_api|common.nativeexec_bindnow (Failed)
	 24 - code_api|common.nativeexec_exe_opt (Failed)
	 25 - code_api|common.nativeexec_bindnow_opt (Failed)
	 67 - code_api|linux.sigcontext (Failed)
	155 - code_api|client.signal (Failed)
	184 - code_api|client.pcache-use (Failed)
	284 - code_api|api.static_sideline_FLAKY (Failed)
```

The failure are likely related to upgrading to gcc 6 and perhaps other pieces of the distro.

### PR DESCRIPTION
**Pull request by hgreving2304**
Disables checklevel option for release build regression.

Fixes #2868

### CODE REVIEW

**Comment by derekbruening** [comment]
Wrong issue number in title and descr?

**Comment by hgreving2304** [comment]
(PTAL)

**Comment by derekbruening** [comment]
Was waiting for the issue number to be fixed

**Comment by hgreving2304** [comment]
(done)

File: suite/tests/CMakeLists.txt [hunk 1]
```diff
@@ -2333,11 +2333,15 @@ if (CLIENT_INTERFACE)
...
 
   if (X86) # Generated code is x86-specific.
+    set(checklevel "")
+    if (DEBUG)
+      set(checklevel "-checklevel 0")
+    endif ()
     tobuild_api(api.ibl-stress api/ibl-stress.c
       # This tests indirect branches between blocks.
       # We use -checklevel 0 to disable the DOCHECK in check_thread_vm_area
       # which makes building 150K bbs very slow.
```
Comment (derekbruening) [inline suite/tests/CMakeLists.txt]: Seems like the comment should go up by the actual instance of -checklevel 0

File: suite/tests/CMakeLists.txt [hunk 2]
```diff
@@ -2333,11 +2333,15 @@ if (CLIENT_INTERFACE)
   endif ()
```
Comment (derekbruening) [inline suite/tests/CMakeLists.txt]: nit: commit title: the text after the i# and before the colon is supposed to be a very short description of what the issue in general covers, with the text after the colon a description of what this particular commit does. 
 The before-colon text should match earlier commits, which say `i#2868 test failures: `.

File: suite/tests/CMakeLists.txt (hunk 1 above)
Comment (hgreving2304) [reply suite/tests/CMakeLists.txt]: Done

File: suite/tests/CMakeLists.txt (hunk 2 above)
Comment (hgreving2304) [reply suite/tests/CMakeLists.txt]: Done

**Pull request merged**